apis/python/src/tiledbsoma/dataframe.py

```diff
@@ -163,6 +163,23 @@ def get_index_column_names(self) -> Sequence[str]:
 
         return self._index_column_names
 
+    @property
```

Comment:
Can you add __len__ per @thetorpedodog 's suggestion?

Reply:
@bkmartinjr I didn't see the suggestion but it's a great one -- thanks!

Reply:
#649